src/Symfony/Component/Validator/Constraints/UniqueValidator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Yevgeniy Zholkevskiy <zhenya.zholkevskiy@gmail.com>
+ */
+class UniqueValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Unique) {
+            throw new UnexpectedTypeException($constraint, Unique::class);
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        if (!\is_array($value) && !$value instanceof \IteratorAggregate) {
+            throw new UnexpectedValueException($value, 'array|IteratorAggregate');
```

Comment:
why not `iterable`?

Reply:
see review chain

Reply:
> I thought so too until recently. Apparently this is considered too unsafe, see [#24577 (comment)](https://github.com/symfony/symfony/pull/24577#issuecomment-358107104) and #25506

?

was it a blocker for this PR?

Reply:
i can pass a broken/unrewindbable iterator as aggregate just as easy. But i cant pass a working iterator/iterable directly. That's poor DX IMHO.

Reply:
Consequencies of this validator being applied on non-rewindable iterators are much bigger than in #25506 so yes (generally, after validator validates the data, data is processed)